From 4f4393feb686aee83acd6aed9c4693dd5f3d61c4 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 24 Jun 2022 15:41:13 +0300
Subject: [PATCH] [SW-91832] gaudi3: erom: add sha files

Added sha.c and sha.h files for gaudi3. This files
are copied from 'arch/gaudi2/common' dir.

This is preparation patch for adding sha.c file
in makefile for compilation for erom_recovery.

Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
Change-Id: Id3fef94bb8154fce21aa26cb0a8f6f95a0c80682
---

diff --git a/arch/gaudi3/common/sha.c b/arch/gaudi3/common/sha.c
new file mode 100644
index 0000000..80dfed8
--- /dev/null
+++ b/arch/gaudi3/common/sha.c
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ *
+ * (C) Copyright 2022 HabanaLabs, Ltd.
+ *
+ */
+
+
+#include <preboot.h>
+#include <uart.h>
+
+#include "hl.h"
+#include <sha.h>
+
+void sha_csum_regions(const struct image_region region[], int region_count,
+			 u8 *output)
+{
+	struct sha512_context ctx;
+	int i;
+
+	sha384_starts(&ctx);
+	for (i = 0; i < region_count; i++)
+		sha384_update(&ctx, region[i].data, region[i].size);
+	sha384_finish(&ctx, output);
+}
diff --git a/arch/gaudi3/common/sha.h b/arch/gaudi3/common/sha.h
new file mode 100644
index 0000000..94b9712
--- /dev/null
+++ b/arch/gaudi3/common/sha.h
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#include <sha512.h>
+#include <preboot/fit_rsa.h>
+
+#define SHA_SUM_LEN			SHA384_SUM_LEN
+#define SHA_DER_LEN			SHA384_DER_LEN
+#define SHA_DER_PRE			sha384_der_prefix
+#define FIT_SHA				"sha384"
+#define FIT_ALGO_PROP_EXPECTED_VALUE	"sha384,rsa3072"
+
+void sha_csum(const unsigned char *input, unsigned int ilen,
+		unsigned char *output);
+
+void sha_csum_regions(const struct image_region region[], int region_count,
+			u8 *output);
